src/main/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActor.scala

```diff
@@ -204,7 +206,27 @@ private class TaskLauncherActor(
     case RecheckIfBackOffUntilReached => OfferMatcherRegistration.manageOfferMatcherStatus()
   }
 
+  @SuppressWarnings(Array("all")) // async/await
   private[this] def receiveTaskLaunchNotification: Receive = {
+    case InstanceOpSourceDelegate.InstanceOpRejected(op, TaskLauncherActor.OfferOperationRejectedTimeoutReason) =>
+      import scala.concurrent.ExecutionContext.Implicits.global
+      // Reschedule instance if provisioning timed out and the instance is not running yet.
+      if (inFlightInstanceOperations.exists(_.instanceId == op.instanceId)) {
+        instanceMap.get(op.instanceId).foreach { instance =>
+          // We don't await the future. It will trigger an instance update event.
+          import org.apache.mesos.Protos
+          val (taskId, _) = instance.tasksMap.head
+          val state = Protos.TaskState.TASK_FAILED
+          val now = clock.now()
+          val phonyFailedTask = Protos.TaskStatus.newBuilder()
```

Comment:
I'll replace this status update once I confirmed that it works.

Reply:
Come to think about it, I would like to keep this phony status. A lot of logic listens to the event bus and it would be non-trivial to introduce a different update. What do you think @timcharper, @alenkacz and @zen-dog?

Reply:
so what are other options? a special event or... ?